File: daily_life/daily_life/report/employee_checkin_script_report/employee_checkin_script_report.js
// Copyright (c) 2023, kalsoom and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Employee Checkin Script Report"] = {
	"filters": [
		{
			"fieldname": "name",
			"label": __("ID"),
			"fieldtype": "Link",
			"options": "Employee Checkin",
			"on_change": function (query_report) {
				const name = frappe.query_report.get_filter_value('name');

				if (name) {
					// Fetch the name of the selected employee
					frappe.call({
						method: "frappe.client.get_value",
						args: {
							doctype: "Employee Checkin",
							filters: { name: name},
							fieldname: "employee_name"
						},
						callback: function (response) {
							const employeeName = response.message["employee_name"];
							// Set the value of the "Employee Name" filter
							frappe.query_report.set_filter_value("employee_name", employeeName);
							frappe.query_report.refresh();
						}
					});
				} else {
					// Clear the "Employee Name" filter if "Employee" filter is cleared
					frappe.query_report.set_filter_value("employee_name", "");
					frappe.query_report.refresh();
				}
			}
		},
		{
			"fieldname": "employee_name",
			"label": __("Employee Name"),
			"fieldtype": "Data",
			"read_only": 1
		},
		{
			"fieldname": "from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.add_days(frappe.datetime.nowdate(), -8),
		},
		{
			"fieldname": "to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.nowdate(),
		}

	]
};

File: daily_life/daily_life/report/employee_checkin_script_report/employee_checkin_script_report.py
# Copyright (c) 2023, kalsoom and contributors
# For license information, please see license.txt




import frappe
from frappe import _

def execute(filters=None):
    if not filters:
        filters = {}
    columns, data = [], []
    conditions = get_conditions(filters)
    columns = get_columns()
    data = get_data(filters, conditions)
    return columns, data

def get_conditions(filters):
    conds = ""
    conds += " and name = %(name)s " if filters.get("name") else ""
    conds += " and DATE(date) between %(from_date)s and %(to_date)s " if filters.get("from_date") and filters.get("to_date") else ""
    return conds

def get_data(filters, conditions):
    data = frappe.db.sql(f"""
        SELECT 
            name,
            employee_name,
            log_type,
            date,
            time,
            designation,
            device_id,
            officein_address
        FROM `tabEmployee Checkin`
        WHERE 1 = 1  
        {conditions}
    """, filters, as_dict=1)

    return data

def get_columns():
    columns = [
        {
            'fieldname': 'name',
            'label': _('Employee'),
            'fieldtype': 'Data',
            'align': 'left',
            'width': 200
        },
        {
            'fieldname': 'employee_name',
            'label': _('Employee Name'),
            'fieldtype': 'Data',
            'align': 'left',
            'width': 200
        },
        {
            'fieldname': 'date',
            'label': _('Date'),
            'fieldtype': 'Data',
            'align': 'left',
            'width': 130
        },
        {
            'fieldname': 'log_type',
            'label': _('Log Type'),
            'fieldtype': 'Data',
            'align': 'left',
            'width': 130
        },
        {
            'fieldname': 'time',
            'label': _('Time'),
            'fieldtype': 'Time',
            'align': 'left',
            'width': 130
        },
        {
            'fieldname': 'designation',
            'label': _('Designation'),
            'fieldtype': 'Data',
            'align': 'left',
            'width': 130
        },
        {
            'fieldname': 'device_id',
            'label': _('Location/Device ID'),
            'fieldtype': 'Data',
            'align': 'left',
            'width': 200
        },
        {
            'fieldname': 'officein_address',
            'label': _('Officein Address'),
            'fieldtype': 'Data',
            'align': 'left',
            'width': 250
        }
    ]

    return columns



















